enterprise/views.py

```diff
@@ -585,6 +585,84 @@ def create_enterprise_course_enrollment(request, consent_record, course_id, lice
         if created:
             track_enrollment('data-consent-page-enrollment', request.user.id, course_id, request.path)
 
+    def _enroll_leaner_in_course(
+            self,
+            request,
+            enterprise_customer,
+            success_url,
+            failure_url,
+            course_id,
+            program_uuid,
+            license_uuid
+    ):
+        """Enrolls an enterprise learner into a course."""
+        # Create EnterpriseCourseEnrollment if we found course_run_id instead of course_key in course_id param.
+        # Skip creating EnterpriseCourseEnrollment if we found course_key instead of course_run_id.
+
+        # EnterpriseCourseEnrollment will be created when the user will select a course run.
+        # A CourseEnrollment record will be created and on the post signal of the CourseEnrollment,
+        # an EnterpriseCourseEnrollment record will also get created.
+        if course_id and self.is_course_run_id(course_id):
+            if license_uuid:
+                enrollment_api_client = EnrollmentApiClient()
+                course_modes = [mode['slug'] for mode in enrollment_api_client.get_course_modes(course_id)]
+                LOGGER.info(
+                    'Retrieved Course Modes for Course {course_id}: {course_modes}'.format(
+                        course_id=course_id,
+                        course_modes=course_modes
+                    )
+                )
+                course_mode = 'verified' if 'verified' in course_modes \
+                    else 'professional' if 'professional' in course_modes \
+                    else 'no-id-professional' if 'no-id-professional' in course_modes \
+                    else 'audit'
+                try:
+                    enrollment_api_client.enroll_user_in_course(
+                        request.user.username,
+                        course_id,
+                        course_mode
+                    )
+                    LOGGER.info(
+                        'Created LMS enrollment for User {user} in Course {course_id} '
+                        'with License {license_uuid} in Course Mode {course_mode}.'.format(
+                            user=request.user.username,
+                            course_id=course_id,
+                            license_uuid=license_uuid,
+                            course_mode=course_mode
+                        )
+                    )
+                except Exception as exc:  # pylint: disable=broad-except
+                    LOGGER.error(
+                        'Unable to create an LMS enrollment from the DSC view: {exc}'.format(
+                            exc=exc
+                        )
+                    )
+                    return redirect(failure_url)
```

Comment:
piggybacking on Alex's comment re: success_url and failure_url not being needed here - I'd avoid catching the exception in here with a redirect since redirecting to a failure URL is an over-opinionated way of handling an enrollment error. Keep this logging in place but re-throw the exception, catch it where the util method is used and avoid having to pass the URLs as args to the utility method